hera/editor/MapEditor.tsx

```diff
@@ -754,6 +869,16 @@ export default function MapEditor({
     }
   }, [saveState]);
 
+  useEffect(() => {
+    if (editor.undoStack) {
+      const stack = editor.undoStack.map(([key, value]) => [
+        key,
+        value.toJSON(),
+      ]);
+      Storage.setItem(getUndoStackKey(mapObject?.id), JSON.stringify(stack));
+    }
+  }, [editor.undoStack, mapObject?.id]);
+
```

Comment:
moved the stack saving to localstorage from updateUndoStack to here because it was out of sync by one